docs/pt/docs/advanced/websockets.md

```diff
@@ -0,0 +1,257 @@
+
+Em seu rota WebSocket você pode `await` por mensagens e enviar mensagens.
+
+```Python hl_lines="48-52"
+{!../../docs_src/websockets/tutorial001.py!}
+```
+
+Você pode receber e enviar dados binários, de texto e JSON.
+
+## Tente você mesmo
+
+Se seu arquivo for nomeado `main.py`, execute sua aplicação com:
+
+<div class="termy">
+
+```console
+$ fastapi dev main.py
+
+<span style="color: green;">INFO</span>:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+</div>
+
+Abra seu broser em: <a href="http://127.0.0.1:8000" class="external-link" target="_blank">http://127.0.0.1:8000</a>.
+
+Você verá uma página simples como:
+
+<img src="/img/tutorial/websockets/image01.png">
+
+Você pode digitar mensagens na caixa de entrada e enviá-las:
+
+<img src="/img/tutorial/websockets/image02.png">
+
+E sua aplicação **FastAPI** com WebSockets responderá de volta:
+
+<img src="/img/tutorial/websockets/image03.png">
+
+Você pode enviar (e receber) muitas mensagens:
+
+<img src="/img/tutorial/websockets/image04.png">
+
+E todas elas usarão a mesma conexão WebSocket.
+
+## Usando `Depends` e outros
+
+Nos endpoints WebSocket você pode importar do `fastapi` e usar:
+
+* `Depends`
+* `Security`
+* `Cookie`
+* `Header`
+* `Path`
+* `Query`
+
+They work the same way as for other FastAPI endpoints/*path operations*:
+Eles funcionam da mesma forma que para outros endpoints FastAPI/*operações de rota*:
+
+//// tab | Python 3.10+
+
+```Python hl_lines="68-69  82"
+{!> ../../docs_src/websockets/tutorial002_an_py310.py!}
+```
+
+////
+
+//// tab | Python 3.9+
+
+```Python hl_lines="68-69  82"
+{!> ../../docs_src/websockets/tutorial002_an_py39.py!}
+```
+
+////
+
+//// tab | Python 3.8+
+
+```Python hl_lines="69-70  83"
+{!> ../../docs_src/websockets/tutorial002_an.py!}
+```
+
+////
+
+//// tab | Python 3.10+ non-Annotated
+
+/// tip | dica
+
+Tente usar a versão `Annotated` se possível.
+
+///
+
+```Python hl_lines="66-67  79"
+{!> ../../docs_src/websockets/tutorial002_py310.py!}
+```
+
+////
+
+//// tab | Python 3.8+ non-Annotated
+
+/// tip | dica
+
+Tente usar a versão `Annotated` se possível.
+
+///
+
+```Python hl_lines="68-69  81"
+{!> ../../docs_src/websockets/tutorial002.py!}
+```
+
+////
+
+/// info
+
+Como isso é um WebSocket, não faz muito sentido levantar uma `HTTPException`, em vez disso levantamos uma `WebSocketException`.
+
+Você pode usar um código de fechamento dos <a href="https://tools.ietf.org/html/rfc6455#section-7.4.1" class="external-link" target="_blank">códigos válidos definidos na especificação</a>.
+
+///
+
+### Tente os WebSockets com dependências
+
+Se seu arquivo for nomeado `main.py`, execute sua aplicação com:
+
+<div class="termy">
+
+```console
+$ fastapi dev main.py
+
+<span style="color: green;">INFO</span>:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+</div>
+
+Abrar seu browser em: <a href="http://127.0.0.1:8000" class="external-link" target="_blank">http://127.0.0.1:8000</a>.
```

Comment:
You can translate browser here (and remove the r in Abra):

Abra o seu navegador em: ...